### PR TITLE
Request JIT accuracy

### PR DESCRIPTION
**Pull request by fabian-guerra**
Adds a new Info.plist property to request accuracy authorization when the accuracy is `CLAccuracyAuthorizationReducedAccuracy` and has authorized location updates.

### CODE REVIEW

File: platform/ios/src/MGLMapView.mm [hunk 1]
```diff
@@ -6098,7 +6103,19 @@ - (void)locationManagerDidChangeAuthorization:(id<MGLLocationManager>)manager
...
+        if (@available(iOS 14, *)) {
+            if ((manager.authorizationStatus != kCLAuthorizationStatusRestricted ||
+                 manager.authorizationStatus != kCLAuthorizationStatusAuthorizedAlways ||
+                 manager.authorizationStatus != kCLAuthorizationStatusAuthorizedWhenInUse) &&
+                manager.accuracyAuthorization == CLAccuracyAuthorizationReducedAccuracy &&
+                [self accuracyDescriptionString] != nil ) {
+                [self.locationManager requestTemporaryFullAccuracyAuthorizationWithPurposeKey:@"MBXAuthorizationAccuracyDescription"];
+            } else {
+                [self validateLocationServices];
+            }
```
Comment (julianrex) [inline platform/ios/src/MGLMapView.mm:6115]: Do we need a `#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000` here?

Comment (fabian-guerra) [reply platform/ios/src/MGLMapView.mm:6115]: Yes.

**Comment by neelmistry94** [comment]
LGTM, no concerns

File: platform/ios/docs/guides/Info.plist Keys.md [hunk 2]
```diff
@@ -10,6 +10,21 @@ Mapbox-hosted vector tiles and styles require an API access token, which you can
...
+Example:
+```
+<key>NSLocationTemporaryUsageDescriptionDictionary</key>
+	<dict>
+		<key>MGLAccuracyAuthorizationDescription</key>
+		<string>Mapbox requires your precise location to help you navigate the map.</string>
+	</dict>
+```
+
+Remove `MGLAccuracyAuthorizationDescription` if you want to control when to request for accuracy authorization.
```
Comment (captainbarbosa) [inline platform/ios/docs/guides/Info.plist Keys.md:26]: I think this section needs more color, as this section feels unclear to me. An example that illustrates this further would be ideal.

File: platform/ios/docs/guides/Info.plist Keys.md [hunk 3]
```diff
@@ -10,6 +10,21 @@ Mapbox-hosted vector tiles and styles require an API access token, which you can
 
 As an alternative, you can use `MGLAccountManager.accessToken` to set a token in code. See [our guide](https://www.mapbox.com/help/ios-private-access-token/) for some tips on keeping access tokens in open source code private.
 
+## MGLAccuracyAuthorizationDescription
+
+Set the Mapbox accuracy authorization description string as an element of `NSLocationTemporaryUsageDescriptionDictionary` to be used by the map to request authorization when the `MGLLocationManager.accuracyAuthorization` is set to `CLAccuracyAuthorizationReducedAccuracy`.
```
Comment (jmkiley) [inline platform/ios/docs/guides/Info.plist Keys.md:15]: Should we note that this is only available in iOS 14+?

**Pull request merged**